src/platform/packages/shared/kbn-esql-utils/src/utils/query_parsing_helpers.ts

```diff
@@ -147,3 +149,35 @@ export const getQueryColumnsFromESQLQuery = (esql: string): string[] => {
 
   return columns.map((column) => column.name);
 };
+/**
+ * This function is used to map the variables to the columns in the datatable
+ * @param esql:string
+ * @param variables:ESQLControlVariable[]
+ * @param columns:DatatableColumn[]
+ * @returns DatatableColumn[]
+ */
+export const mapVariableToColumn = (
+  esql: string,
+  variables: ESQLControlVariable[],
+  columns: DatatableColumn[]
+): DatatableColumn[] => {
+  if (!variables.length) {
+    return columns;
+  }
+  const { root } = parse(esql);
+  const usedVariablesInQuery = Walker.params(root).map((param) => param);
```

Comment:
Why map here if the callback is just returning the identity?

Reply:
Very good catch https://github.com/elastic/kibana/pull/202875/commits/cbc7001f9171397083ba4f847f94cc07c3af472e